notebooks/py/assay_comparison.py

```diff
@@ -0,0 +1,237 @@
+spots(decoded_image=merfish_decoded_image, background_image=merfish_dots, decoded_image_kwargs=dict(alpha=1.), ax=ax4)
+spots(decoded_image=merfish_decoded_image, background_image=merfish_nuclei, decoded_image_kwargs=dict(alpha=1.), ax=ax6);
+# EPY: END code
+
+# EPY: START markdown
+#From these examples, we can see that the point cloud over-estimates the spot size (perhaps we're calculating radius wrong?)
+# EPY: END markdown
+
+# EPY: START markdown
+### Show different Dartfish Spots Plots
+# EPY: END markdown
+
+# EPY: START code
+f, ((ax1, ax2), (ax3, ax4), (ax5, ax6)) = plt.subplots(ncols=2, nrows=3, figsize=(30, 45))
+spots(dartfish, background_image=np.zeros_like(dartfish_dots), spots_kwargs=dict(alpha=1.), ax=ax1)
+spots(dartfish, background_image=dartfish_dots, spots_kwargs=dict(alpha=1.), ax=ax3)
+spots(dartfish, background_image=dartfish_nuclei, spots_kwargs=dict(alpha=1.), ax=ax5)
+spots(decoded_image=dartfish_decoded_image, decoded_image_kwargs=dict(alpha=1.), ax=ax2)
+spots(decoded_image=dartfish_decoded_image, background_image=dartfish_dots, decoded_image_kwargs=dict(alpha=1.), ax=ax4)
+spots(decoded_image=dartfish_decoded_image, background_image=dartfish_nuclei, decoded_image_kwargs=dict(alpha=1.), ax=ax6);
+# EPY: END code
+
+# EPY: START markdown
+### Show different background types for ISS
+# EPY: END markdown
+
+# EPY: START code
+f, ((ax1, ax2), (ax3, ax4), (ax5, ax6)) = plt.subplots(ncols=2, nrows=3, figsize=(30, 40))
+spots(iss, background_image=np.zeros_like(iss_dots), spots_kwargs=dict(alpha=1.), ax=ax1)
+spots(iss, background_image=iss_dots, spots_kwargs=dict(alpha=1.), ax=ax3)
+spots(iss, background_image=iss_nuclei, spots_kwargs=dict(alpha=1.), ax=ax5)
+
+# ISS doesn't have a decoded image, so leave those plots blank to make this clear
+for ax in (ax2, ax4, ax6):
+    ax.set_axis_off()
+f.tight_layout()
+# EPY: END code
+
+# EPY: START markdown
+### Download available copy number information from original authors
+# EPY: END markdown
+
+# EPY: START code
+dartfish_copy_number = pd.read_csv('https://dmf0bdeheu4zf.cloudfront.net/20180821/DARTFISH/fov_001/counts.csv', index_col=0, squeeze=True)
+merfish_copy_number = pd.read_csv(data_root + "merfish_copy_number_benchmark.csv", index_col=0, squeeze=True)
+iss_copy_number = pd.read_csv(data_root + "iss_copy_number_benchmark.csv", index_col=1)['cnt']
+# EPY: END code
+
+# EPY: START code
+f, axes = plt.subplots(ncols=3, nrows=4, figsize=(12, 14))
+
+for dataset, benchmark, axis in zip([iss, merfish, dartfish], [iss_copy_number, merfish_copy_number, dartfish_copy_number], axes[0, :]):
+    compare_copy_number(dataset, benchmark, ax=axis, color='tab:blue')
+
+for dataset, axis in zip(datasets, axes[1, :]):
+    barcode_magnitude_histogram(dataset, bins=20, ax=axis)
+
+for dataset, axis in zip(datasets, axes[2, :]):
+    spot_area_histogram(dataset, bins=20, ax=axis)
+
+for dataset, axis in zip([merfish, dartfish], axes[3, 1:]):
```

Comment:
we should be able to plot the spot distance histogram for iss. all the information to compute this metric is in the intensity table. as such, we should plot it. we will also learn something about our API in constructuing this figure. 

additionally, in the olden days, i had a quality metric for iss spot detection but i think it was lost. it was a good metric for the per_round_max_channel_decoder. was this code nuked? if so, why?

Reply:
also, it's probably worth generalizing spot_distance_histogram to spot_quality_histogram. for some assays (e.g., pixel read out ones) the quality will be distance. for others, it will be some other quality score thats particular to the decoder and/or chemistry (e.g., the old iss quality metric)

Reply:
I wonder if we just want to generalize "histogram" to have some basic style parameters and an optional threshold.

Reply:
> we should be able to plot the spot distance histogram for iss. all the information to compute this metric is in the intensity table. as such, we should plot it. we will also learn something about our API in constructuing this figure.

Actually, master does _not_ have the required code. My filtering branch rectifies this by reporting the fraction of the signal for each round that resides in the selected channel, however currently the distance is 1 if the spot fails to decode, and zero otherwise. 

We should circle back and update this when that fix is made.